src/test/java/stringcalculator/StringCalculator.java

```diff
@@ -0,0 +1,90 @@
+package stringcalculator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringCalculator {
+    private static final String OPERATION_REGEX = "[+\\-*/]+";
+    private static final String NUMBER_REGEX = "\\d+";
+    private static final String NOT_VALID_REGEX = "[^\\d+\\-*/^\\s]";
+
+    public static int calculate(String input) {
+        containsOnlyValidCharacters(input);
+        List<Integer> numbers = toInts(getCharacters(input, NUMBER_REGEX));
+        List<String> operations = getCharacters(input, OPERATION_REGEX);
+        int result = numbers.get(0);
+        for (int i = 0; i < operations.size(); i++) {
+            result = operate(numbers, operations, result, i);
+        }
+        return result;
+    }
+
+    private static int operate(List<Integer> numbers, List<String> operations, int result, int i) {
+        result = plus(numbers, operations, result, i);
+        result = minus(numbers, operations, result, i);
+        result = divide(numbers, operations, result, i);
+        result = multiple(numbers, operations, result, i);
+        return result;
+    }
+
+    private static int multiple(List<Integer> numbers, List<String> operations, int result, int i) {
+        if (operations.get(i).equals("/")) {
+            result /= numbers.get(i + 1);
+        }
+        return result;
+    }
+
+    private static int divide(List<Integer> numbers, List<String> operations, int result, int i) {
+        if (operations.get(i).equals("*")) {
+            result *= numbers.get(i + 1);
+        }
+        return result;
+    }
+
+    private static int minus(List<Integer> numbers, List<String> operations, int result, int i) {
+        if (operations.get(i).equals("-")) {
+            result -= numbers.get(i + 1);
+        }
+        return result;
+    }
+
+    private static int plus(List<Integer> numbers, List<String> operations, int result, int i) {
+        if (operations.get(i).equals("+")) {
+            result += numbers.get(i + 1);
+        }
+        return result;
+    }
+
+    private static List<String> getCharacters(String input, String regex) {
+        List<String> operations = new ArrayList<>();
+
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(input);
+
+        while (matcher.find()) {
+            String op = matcher.group();
```

Comment:
이번 과정에서는 축약은 지양해주세요~ 
명확한 의미 전달이 중요하다고 생각합니다!

```suggestion
            operations.add(matcher.group());
```

별다른 사용하는 변수가 아니니 바로 인자로 넘겨줘도 좋겠네요 🤔